### PR TITLE
Update c++ readers doc

### PR DESCRIPTION
**Pull request by JiayiFeng**
This PR refines C++ readers doc according to @helinwang's comments of #9079:

1. Rename a section title from `CreateReaderOp` to `Operators That Creates Readers`.

2. Add the introduction of `MultiPassReader` and how to simplify program configurations with it.

### CODE REVIEW

File: doc/fluid/design/concepts/cpp_data_feeding.md [hunk 1]
```diff
@@ -153,13 +153,17 @@ double_buffer_reader = create_double_buffer_op(batch_reader)
 The forwarding ops of the corresponding `main_program` would be like this:
 
 ```
-while_op {
+not_completed = true
+pass_count = 0
```
Comment (helinwang) [inline doc/fluid/design/concepts/cpp_data_feeding.md]: `pass_count` is a Python variable, not a ProgramDesc Var, does it work?

File: doc/fluid/design/concepts/cpp_data_feeding.md [hunk 2]
```diff
@@ -169,3 +173,30 @@ Two important considerations for these programs are as follows:
 1. The multiple\_reader is the batch\_reader's underlying reader, and the batch\_reader is the double\_buffer\_reader's underlying reader. `read_op`, `has_next_op` and other reader related ops will only invoke the top-most reader. In this case, it's the double\_buffer\_reader.
 
 2. All readers exist in both `startup_program` and `main_program`. And they are persistable.
+
+### Simplify Configuration by MultiPassReader
+
+The  Program configuration mentioned above is somehow complicated. Users need to be very similar to concepts of Program and Block to prevent making mistakes in their code. To make the usage of C++ readers more friendly to beginning users, we introduce `MultiPassReader`.
```
Comment (helinwang) [inline doc/fluid/design/concepts/cpp_data_feeding.md]: The  Program -> The Program (please change two spaces to one space).
similar -> familiar

File: doc/fluid/design/concepts/cpp_data_feeding.md [hunk 3]
```diff
@@ -169,3 +173,30 @@ Two important considerations for these programs are as follows:
...
+batch_reader = create_batch_reader_op(multiple_reader)
+multi_pass_reader = create_multi_pass_reader_op(batch_reader)
+double_buffer_reader = create_double_buffer_op(multi_pass_reader)
+... (other initializers)
+```
+
+The forwarding part of the corresponding `main_program` would be like this:
+
+```
+not_completed = true
```
Comment (helinwang) [inline doc/fluid/design/concepts/cpp_data_feeding.md]: `not_completed ` is a Python variable, not a ProgramDesc Var, does it work?

File: doc/fluid/design/concepts/cpp_data_feeding.md (hunk 1 above)
Comment (JiayiFeng) [reply doc/fluid/design/concepts/cpp_data_feeding.md]: `not_completed` and `pass_count` are both ProgramDesc Var. Sorry I didn't make it clear.

File: doc/fluid/design/concepts/cpp_data_feeding.md (hunk 2 above)
Comment (JiayiFeng) [reply doc/fluid/design/concepts/cpp_data_feeding.md]: Done. Thanks!

File: doc/fluid/design/concepts/cpp_data_feeding.md (hunk 3 above)
Comment (JiayiFeng) [reply doc/fluid/design/concepts/cpp_data_feeding.md]: They are Fluid Variables. See my above comment.

**Pull request merged**